Content.Server/Explosion/EntitySystems/TriggerSystem.Mobstate.cs

```diff
@@ -18,12 +18,12 @@ private void InitializeMobstate()
 
     private void OnMobStateChanged(EntityUid uid, TriggerOnMobstateChangeComponent component, MobStateChangedEvent args)
     {
-        if (component.MobState != args.NewMobState)
+        if (!component.MobState.Contains(args.NewMobState))
             return;
 
-        //This chains Mobstate Changed triggers with OnUseTimerTrigger if they have it
+        //This chains Mobstate Changed triggers with OnUseTimerTrigger if they have it and if the mob is dead
         //Very useful for things that require a mobstate change and a timer
-        if (TryComp<OnUseTimerTriggerComponent>(uid, out var timerTrigger))
+        if (TryComp<OnUseTimerTriggerComponent>(uid, out var timerTrigger) && args.NewMobState == MobState.Dead)
         {
             HandleTimerTrigger(
                 uid,
```

Comment:
This should still be checking for mobstate and not if the mobstate is dead. There could be entities out there using this and checking for some other value that's not dead.

Reply:
It only checks if the MobState is dead for the followup of a TimerTrigger, which is currently only used by the macrobomb and the new deathrattle one which both trigger it on death. If I remove the mobstate death check it will trigger all effects of an implant that supplies multiple MobStates on every given state.

Reply:
The issue is if you make it locked to on Death then this becomes a singleton method and won't be open for potentially more interesting things (OnCrit, etc). So I'm standing firm behind no, this should be reverted back to its original state.

Reply:
I already removed it on the last commit